test/cpp/qps/server_async.cc

```diff
@@ -166,8 +100,16 @@ class AsyncQpsServerTest {
         return;
       }));
     }
-    while (!got_sigint) {
-      std::this_thread::sleep_for(std::chrono::seconds(5));
+  }
+  ~AsyncQpsServerTest() {
+    server_->Shutdown();
+    srv_cq_.Shutdown();
```

Comment:
The child threads return when they see the shutdown event. I'm looking at the underlying code in core, though, and not seeing where the cq will prohibit things from being added after a shutdown has been requested. Am I missing something?